ts/features/bonus/cgn/components/merchants/CgnMerchantListItem.tsx

```diff
@@ -26,6 +24,62 @@ const styles = StyleSheet.create({
     paddingVertical: 16
   }
 });
+const renderCategoryElement = (category: ProductCategory) =>
+  getCategorySpecs(category).fold(undefined, c => (
+    <View style={styles.row}>
+      {c.icon({ height: 22, width: 22, fill: IOColors.bluegrey })}
+      <View hspacer small />
+      <H5 weight={"SemiBold"} color={"bluegrey"}>
+        {I18n.t(c.nameKey).toLocaleUpperCase()}
+      </H5>
+    </View>
+  ));
+
+// This is used to render the product categories of a merchant
+// if there's more than 2 categories available we render a placeholder text to enumerate how many categories
+// are missing from the list
+const CategoriesRow = ({ categories }: Pick<Props, "categories">) => (
+  <>
+    {categories.length > 2
```

Comment:
Not sure server-side is possible, but if there are 3 categories and 2 of them are the same this check doesn't work as expected:
<img src="https://user-images.githubusercontent.com/11773070/135587982-493ff955-6bf8-4bb6-8930-ced6469946ed.png" width=300>
<img src="https://user-images.githubusercontent.com/11773070/135588101-57f18624-9ca4-457f-98d0-ec48f947e9a4.png" width=300>

Reply:
@debiff This mostly depends on data received, I would avoid any logic to avoid duplicated on the client. Anyway this should not be possible by backend API